### PR TITLE
feat: add tupctl tool

### PR DESCRIPTION
**Pull request by azanelli-actronika**
which enable to user to control a tactronik from a command line.

### CODE REVIEW

File: tools/tupctl.c [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+    { 'h', "help", "Show this help", NULL, print_help_and_exit },
+};
+
+static void print_help_and_exit(void)
+{
+    int i;
+
+    printf("Available commands:\n");
+    for (i = 0; i < N_ELEMENTS(cmds); i++) {
+        printf("  %-16s %-40s%s\n", cmds[i].name, cmds[i].args_desc, cmds[i].desc);
```
Comment (JeremyActronika) [inline tools/tupctl.c]: More than 80 characters

File: tools/tupctl.c [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+    effect_id = atoi(argv[0]);
+
+    printf("Stopping effect in slot %d\n", effect_id);
+    tup_message_init_stop(&msg, effect_id);
+    ret = tup_context_send(&tup_ctx, &msg);
+    tup_message_clear(&msg);
+    return ret;
+}
+
+static int do_get_version(int argc, char *argv[])
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: not used so...

File: tools/tupctl.c [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+    int loaded_id;
+    int effect_id;
+    int ret;
+
+    if (argc != 2) {
+        printf("'load' arguments: <slot-id> <effect-id>\n");
+        return -EINVAL;
+    }
+
+    loaded_id = atoi(argv[0]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+    effect_id = atoi(argv[1]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+    TupMessage msg;
+    int effect_id;
+    int ret;
+
+    if (argc != 1) {
+        printf("'play' arguments: <slot-id>\n");
+        return -EINVAL;
+    }
+
+    effect_id = atoi(argv[0]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 6]
```diff
@@ -0,0 +1,479 @@
...
+    TupMessage msg;
+    int effect_id;
+    int ret;
+
+    if (argc != 1) {
+        printf("'stop' arguments: <slot-id>\n");
+        return -EINVAL;
+    }
+
+    effect_id = atoi(argv[0]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 7]
```diff
@@ -0,0 +1,479 @@
...
+    int effect_id, parameter_id;
+    int ret;
+
+    if (argc != 2) {
+        printf("'get_parameter' arguments: <slot-id> <parameter-id>\n");
+        return -EINVAL;
+    }
+
+    effect_id = atoi(argv[0]);
+    parameter_id = atoi(argv[1]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 8]
```diff
@@ -0,0 +1,479 @@
...
+    int ret;
+
+    if (argc != 3) {
+        printf("'set_parameter' arguments: <slot-id> <parameter-id> <value>\n");
+        return -EINVAL;
+    }
+
+    effect_id = atoi(argv[0]);
+    parameter_id = atoi(argv[1]);
+    parameter_value = atoi(argv[2]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 9]
```diff
@@ -0,0 +1,479 @@
...
+        printf("'bind_effect' arguments: <slot-id> <binding-flags>\n"
+                "binding-flags: 0 -> unbind\n"
+                "               1 -> actuator 1\n"
+                "               2 -> actuator 2\n"
+                "               3 -> actuator 1 and 2\n");
+        return -EINVAL;
+    }
+
+    effect_id = atoi(argv[0]);
+    flags = atoi(argv[1]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 10]
```diff
@@ -0,0 +1,479 @@
...
+    TupMessage msg;
+    int sensor_id;
+    int ret;
+
+    if (argc != 1) {
+        printf("'get_sensor_value' arguments: <sensor-id>");
+        return -EINVAL;
+    }
+
+    sensor_id = atoi(argv[0]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c [hunk 11]
```diff
@@ -0,0 +1,479 @@
...
+    int sensor_value;
+    int ret;
+
+    if (argc != 1) {
+        printf("'set_sensor_value' arguments: <sensor-id> <value>");
+        return -EINVAL;
+    }
+
+    sensor_id = atoi(argv[0]);
+    sensor_value = atoi(argv[1]);
```
[the remark this answers is not in the document]
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

File: tools/tupctl.c (hunk 1 above)
Comment (azanelli-actronika) [reply tools/tupctl.c]: done

**Comment by azanelli-actronika** [comment]
> Some warning (look below)

I fixed the one introduced in tupctl, warnings was not activated in my workspace (sadly).

**Pull request merged**